### PR TITLE
chore: Migrate to eslint

### LINKED ISSUE #11237: Migrate from tslint to eslint
As `tslint` is deprecated for some time already we should also move to `eslint` with angular/typescript plugins.

### PR DESCRIPTION
**Pull request by marlass**
What was done:
- run the migrations with the angular-eslint
- adjusted configs in packages to prevent any errors at the moment
- change projects setup in eslint for performance
- remove tslint dependencies and configs

Next steps for future PR:
- enable new rules that we think are worth to have

Closes #11237

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -124,6 +124,11 @@
     "@angular-devkit/build-ng-packagr": "~0.1002.1",
     "@angular-devkit/core": "10.2.1",
     "@angular-devkit/schematics": "^10.2.1",
+    "@angular-eslint/builder": "1.2.0",
+    "@angular-eslint/eslint-plugin": "1.2.0",
+    "@angular-eslint/eslint-plugin-template": "1.2.0",
+    "@angular-eslint/schematics": "1.2.0",
+    "@angular-eslint/template-parser": "1.2.0",
```
Comment (Pucek9) [inline package.json]:
```suggestion
    "@angular-eslint/builder": "^1.2.0",
    "@angular-eslint/eslint-plugin": "^1.2.0",
    "@angular-eslint/eslint-plugin-template": "^1.2.0",
    "@angular-eslint/schematics": "^1.2.0",
    "@angular-eslint/template-parser": "^1.2.0",
```

File: package.json [hunk 2]
```diff
@@ -150,6 +156,11 @@
     "ejs": "^2.6.2",
     "enquirer": "^2.3.6",
     "env-cmd": "^10.1.0",
+    "eslint": "^7.6.0",
+    "eslint-plugin-deprecation": "^1.2.0",
+    "eslint-plugin-import": "2.22.1",
+    "eslint-plugin-jsdoc": "30.7.6",
+    "eslint-plugin-prefer-arrow": "1.2.2",
```
Comment (Pucek9) [inline package.json]:
```suggestion
    "eslint-plugin-deprecation": "^1.2.0",
    "eslint-plugin-import": "^2.22.1",
    "eslint-plugin-jsdoc": "^30.7.6",
    "eslint-plugin-prefer-arrow": "1.2.2",
```

File: package.json [hunk 3]
```diff
@@ -140,8 +145,9 @@
     "@types/parse5": "^5.0.3",
     "@types/semver": "6",
     "@types/shelljs": "^0.8.7",
+    "@typescript-eslint/eslint-plugin": "4.3.0",
+    "@typescript-eslint/parser": "4.3.0",
```
Comment (Pucek9) [inline package.json]:
```suggestion
    "@typescript-eslint/eslint-plugin": "^4.3.0",
    "@typescript-eslint/parser": "^4.3.0",
```

**Review by Pucek9 — CHANGES_REQUESTED** [review]
IMO we could simplify particular `.eslintrc.json` files to avoid duplicate configuration for overriding exactly the same stuff

File: package.json [hunk 4]
```diff
@@ -150,6 +156,11 @@
     "ejs": "^2.6.2",
     "enquirer": "^2.3.6",
     "env-cmd": "^10.1.0",
+    "eslint": "^7.6.0",
+    "eslint-plugin-deprecation": "^1.2.0",
+    "eslint-plugin-import": "^2.22.1",
+    "eslint-plugin-jsdoc": "^30.7.6",
+    "eslint-plugin-prefer-arrow": "1.2.2",
```
Comment (marlass) [inline package.json]:
```suggestion
    "eslint-plugin-prefer-arrow": "^1.2.2",
```

**Review by Pucek9 — APPROVED** [review]

**Review by dunqan — APPROVED** [review]

**Pull request merged**